docs/book/v1/intro.md

```diff
@@ -1,3 +1,37 @@
-# laminas-{component}
+# laminas-servicemanager-migration
+
+### Using with Rector
+
+To use with rector, you can create the following `rector.php`:
```

Comment:
We could require (`require-dev`) `laminas-cli` in here, create a command like `servicemanager:migration:init` and let that command create that `rector.php`?

We could refer to that command aswell and thus we have a maximum convenient way to migrate.

Reply:
Maybe its even possible to have `servicemanager:migration:run` which handles that whole configuration stuff so there is no need to create any configuration in the project at all.

As I can see, the only thing the user has to provide is the directories where to run the migration on. We could do that in the command aswell.

What do you think about this?

Reply:
/cc @TomasVotruba this is the repository in laminas org, is there a way to call `internal` rector `refactor` method outside the rector command to avoid run command to run another command? We need to do like this:

```bash
vendor/bin/laminas service manager:migration:run
```

that run the rector with rector config.

Reply:
You can re-use `ContainerFactory` and recreate `GenerateCommand::execute()` method yourself.
Not sure if worth it, as you'd have to all the maintenance of future version yourself

Reply:
@TomasVotruba thank you 👍

Reply:
We could atleast automate the `rector.php` generation with a command. 
executing rector could be a manual run or we are using the same approach as with composer [here](https://github.com/laminas/laminas-dependency-plugin/blob/8e422b2a22638bb1d5a55be5a2a9d6cce27ac5dc/src/DependencyRewriterV2.php#L240)

But just to understand, as you've said this multiple times now.
Yes, we have to maintain this by ourselves. But are you planning to introduce that much of bc breaks that you would recommend to not go this way?
If so, if there are BC breaks, we would have to update documentation anyways as today it might be `rector.php` which returns a callable and tomorrow its an object (like in php-cs-fixer)?

So either way, we end up having maintenance anyway, right?

Reply:
BC breaks are comming with every major PHP version. Also with architecture change - as time goes by, better patterns may appear. There are many weak parts inside Rector code base, that should not be copy-pasted.

In general, it's easier to maintain external API - e.g. rule and configured them -, instead of copy-pasted internal classes.